src/ir_York.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Daniele Gobbetti
+
+/// @file
+/// @brief Support for the York AC protocol (remote GRYLH2A)
+
+// Supports:
+//   Brand: York,  Model: MHH07P17
+
+#ifndef IR_YORK_H_
+#define IR_YORK_H_
+
+#define __STDC_LIMIT_MACROS
+#include <stdint.h>
+#ifndef UNIT_TEST
+#include <Arduino.h>
+#endif
+#include "IRremoteESP8266.h"
+#include "IRsend.h"
+#ifdef UNIT_TEST
+#include "IRsend_test.h"
+#endif
+
+
+/// Native representation of a York A/C message.
+union YorkProtocol{
+  uint8_t raw[kYorkStateLength];  ///< The state of the IR remote.
+  struct {
+    // byte 0-5
+    uint8_t preamble[6];  // unknown, fixed 0x08, fixed 0x10, fixed 0x07, fixed 0x02, fixed 0x40, fixed 0x08
+    // byte 6
+    uint8_t Key1     :4;  // key pressed on the remote: 1 power, 2 temp up, 3 temp down...
+    uint8_t Key2     :4;  // only set when setting ontime/offtime: Key1 value is 0x6 (enter key) and Key2 is 0x3 for "start" and 0x2 for "stop"
+    //byte 7
+    uint8_t Fan      :4;  // Fan speed: 1 low, 2 mid, 3 max, 8 auto
+    uint8_t Power    :4;  // main unit power: 1 on, 0 off
+    //byte 8
+    uint8_t Mode     :8;  // 1 heat, 2 cool, 3 dry, 4 fan, 8 auto
```

Comment:
If those are the only possible values, then it is likely that `Mode` is only 4 bits.
so use:
```c++
    uint8_t Mode     :4;  // 1 heat, 2 cool, 3 dry, 4 fan, 8 auto
    uint8_t          :4;
```